### DISCUSSION TITLE
Where can I find the intra-only config file for VVEnc equivalent to VTM's intra-only configuration?

### DISCUSSION THREAD

**Discussion by eimran-eimon** [fraunhoferhhi/vvenc#114] [General]
In this repository, only random access configuration files with different preset are provided. But there is no config file present for intra-only configuration. Can anyone provide me with the intra-only config file for the VVenc?

Thanks.

**Comment by adamjw24** [comment]
There is no intra-only configuration files provided for intra-only encoding. You can use the RA configuration files and just add the following options to your call `--IntraPeriod=1 --GOPSize=1 --MaxParallelFrames=1`, e.g. in a following call:

`vvencFFapp -i input.yuv -c sequence.cfg -c rancomaccess_medium.cfg --IntraPeriod=1 --GOPSize=1 --MaxParallelFrames=1`

Motion compensation and appropriate tools will not be used, so theres nothing to worry about.

We do not calculate a intra-only pareto set, so the presets will not be optimal, but should provide a good quality/runtime scaling.

Since it is not officially supported or optimal, it is not possible to do this using the simple app.

See the following paper some more info:
* A. Wieckowski, C. Stoffers, B. Bross, D. Marpe, "VVenC, an open optimized VVC encoder, in versatile application scenarios," Proc. SPIE 11842, Applications of Digital Image Processing XLIV, 118420H (1 August 2021), doi: 10.1117/12.2595360.

**Reply by eimran-eimon** [reply]
Thanks for your prompt response.

So, we only have to pass these three parameters --IntraPeriod=1 --GOPSize=1 --MaxParallelFrames=1 along with the main command to make the config file equivalent to the VTM's intra-only config, or do we have to change any other config parameters? What about the "DecodingRefreshType" parameter? Should we keep it the same?

Suppose we choose to use the "randomaccess-slower.cfg" by changing those parameters as mentioned above. Could we expect that the bit-per-pixel (bpp) of the encoded file and PSNR will be the same as if we have to use the VTM encoder with the intra-only config? What about the speed-up of the VVEnc in that case? Will it be always faster than the VTM?

**Reply by adamjw24** [reply]
The DecodingRefreshType parameter is free for you to choose. If you keep it at default you'll get an IDR picture followed by CRA pictures, just as VTM does. Setting it `2`, will give you an IDR picture followed by IDR_W_RASL (but of course without the actual RASL pictures). There's currently no option to produce IDR pictures only.

About the bits-per-pixel and PSNR, I cannot really tell. Please have a look at the graph below, which shows the performance of VVenC 1.0 and VTM in the all-intra config. VVenC is always faster than VTM, but not by such a high margin as with random access. The BDR is a combined metric describing both the rate and the quality (bpp and PSNR), so I cannot tell if both will be the same. But combined as an expected rate saving vs HM, VVenC slower has very similar performance to VTM.

![image](https://user-images.githubusercontent.com/70575289/144298472-1e871fcd-0542-4037-9ba5-0d0be7023174.png)
